modules/gapi/test/own/mat_tests.cpp

```diff
@@ -384,4 +384,21 @@ TEST(OwnMat, ROIView)
     << to_ocv(roi_view) << std::endl
     << expected_cv_mat  << std::endl;
 }
+
+TEST(OwnMat, CreateNegMat)
+{
+    // cv::Mat for example. It throws exception in setSize method
+    cv::Mat ocv_mat;
+    EXPECT_THROW(ocv_mat.create(cv::Size(-1,-1), CV_8U), std::exception);
+
+    Mat own_mat_t1, own_mat_t2, own_mat_t3;
+    EXPECT_THROW(own_mat_t1.create(cv::Size{-1, -1}, CV_8U), std::exception);
+    EXPECT_THROW(own_mat_t2.create(cv::Size{ 1, -1}, CV_8U), std::exception);
+    EXPECT_THROW(own_mat_t3.create(cv::Size{-1,  1}, CV_8U), std::exception);
+
+    // if Size{cols, rows} of create method is equal to Size{cols, rows} of Mat case
+    Mat own_mat_t4(-1, -1, CV_8U, nullptr);
+    EXPECT_THROW(own_mat_t4.create(-1, -1, CV_8U), std::exception);
```

Comment:
What if just one of dimensions is negative? Can I create a Mat with zero height?

Reply:
You can't create own::Mat if just one of dimensions is negative.

And you can create own::Mat with zero height.

Reply:
> You can't create own::Mat if just one of dimensions is negative.

My point was that this needs to be demonstrated by tests

> And you can create own::Mat with zero height.

What's the reason in a Mat with zero height? Can we also create a cv::Mat with zero height?

Reply:
We can create cv::Mat with zero dimensions. We can use constructor or create method for this.
But cv::Mat with zero dimensions isn't use anywhere. So create own::Mat with zero dimensions doesn't make sense.

Should we throw an exception in own::Mat's constructor for this case ( rows or cols <= 0 )? It destroys the own::Mat with zero-height creation bug.

Reply:
ok got it